hazelcast/src/main/java/com/hazelcast/config/DynamicClassloadingConfig.java

```diff
@@ -0,0 +1,235 @@
+/*
+ * Copyright (c) 2008-2016, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.config;
+
+/**
+ * Configuration of dynamic classloading. When enabled it allows members to load classes from other members.
+ * This simplifies deployment as you do not have to deploy your domain classes into classpath of all cluster members.
+ *
+ */
+public class DynamicClassloadingConfig {
+
+    /**
+     * Controls caching of caches loaded from remote members
+     */
+    public enum ClassCacheMode {
```

Comment:
I'm still thinking about it. Having it an internal class is somewhat clumsy/versbose. It's not too different from e.g. https://github.com/jerrinot/hazelcast/blob/eef987b9920f3f7b4c506e433b0d067cab9ba93c/hazelcast/src/main/java/com/hazelcast/config/EvictionPolicy.java#L21-L21 @mdogan WDYT?